pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1921,11 +1923,53 @@ public long getLastDisconnectedTimestamp() {
         return connectionHandler.lastConnectionClosedTimestamp;
     }
 
-    private ByteBuf decryptPayloadIfNeeded(MessageIdData messageId, int redeliveryCount, MessageMetadata msgMetadata,
-                                           ByteBuf payload, ClientCnx currentCnx) {
+    /**
+     * Represents the outcome of a message decryption attempt for the consumer.
+     */
+    private static class DecryptResult {
+        private final boolean success;
+        private final ByteBuf decryptedPayload;
```

Comment:
Is it better to use payload instead of decryptedPayload? Since it could be raw data when `success == true`